static/app/views/settings/featureFlags/index.tsx

```diff
@@ -0,0 +1,201 @@
+import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
+import TextBlock from 'sentry/views/settings/components/text/textBlock';
+import {OrganizationFeatureFlagsProviderRow} from 'sentry/views/settings/featureFlags/organizationFeatureFlagsProviderRow';
+
+export type Secret = {
+  createdAt: string;
+  createdBy: number;
+  id: number;
+  provider: string;
+  secret: string;
+};
+
+type FetchSecretResponse = {data: Secret[]};
+
+type FetchSecretParameters = {
+  orgSlug: string;
+};
+
+type RemoveSecretQueryVariables = {
+  id: number;
+};
+
+export const makeFetchSecretQueryKey = ({orgSlug}: FetchSecretParameters) =>
+  [`/organizations/${orgSlug}/flags/signing-secrets/`] as const;
+
+function SecretList({
+  secretList,
+  isRemoving,
+  removeSecret,
+}: {
+  isRemoving: boolean;
+  secretList: Secret[];
+  removeSecret?: (data: {id: number}) => void;
+}) {
+  return (
+    <Fragment>
+      {secretList.map(secret => {
+        return (
+          <OrganizationFeatureFlagsProviderRow
+            key={secret.id}
+            secret={secret}
+            isRemoving={isRemoving}
+            removeSecret={removeSecret ? () => removeSecret({id: secret.id}) : undefined}
+          />
+        );
+      })}
+    </Fragment>
+  );
+}
+
+export function OrganizationFeatureFlagsIndex({
+  organization,
+}: {
+  organization: Organization;
+}) {
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const {
+    isPending,
+    isError,
+    data: secretList,
+    refetch: refetchSecretList,
+  } = useApiQuery<FetchSecretResponse>(
+    makeFetchSecretQueryKey({orgSlug: organization.slug}),
+    {
+      staleTime: Infinity,
+    }
+  );
+
+  const {mutate: handleRemoveSecret, isPending: isRemoving} = useMutation<
+    {},
+    RequestError,
+    RemoveSecretQueryVariables
+  >({
+    mutationFn: ({id}) =>
+      api.requestPromise(
+        `/organizations/${organization.slug}/flags/signing-secrets/${id}/`,
+        {
+          method: 'DELETE',
+        }
+      ),
+
+    onSuccess: (_data, {id}) => {
+      addSuccessMessage(
+        t('Removed the provider and signing secret for the organization.')
+      );
+
+      setApiQueryData(
+        queryClient,
+        makeFetchSecretQueryKey({orgSlug: organization.slug}),
+        (oldData: FetchSecretResponse) => {
+          return {data: oldData.data.filter(oldSecret => oldSecret.id !== id)};
+        }
+      );
+    },
+    onError: error => {
+      const message = t('Failed to remove the provider or signing secret.');
+      handleXhrErrorResponse(message, error);
+      addErrorMessage(message);
+    },
+  });
+
+  const addNewProvider = (
+    <LinkButton
+      priority="primary"
+      size="sm"
+      to={`/settings/${organization.slug}/feature-flags/new-provider/`}
+      data-test-id="create-new-provider"
+    >
+      {t('Add New Provider')}
+    </LinkButton>
+  );
+
+  return (
+    <Access access={['org:write']}>
```

Comment:
Do we also need `org:write` to create secrets? 

I think yes. Currently the mode is like "one secret per provider -> override the value on POSt". Someone with `org:write` access could delete the secret no problem, but we should check for that access level when we do the POST as well, otherwise someone could write `aaaaaaaaa` and basically zero-out the token creating a denial-of-service situation for the org.